doc/developers-guide/accumulators.md

```diff
@@ -164,29 +143,12 @@ Eventually there are three things that may happen:
 Stanzas are often checked against privacy lists.
 According to the current `mongoose_privacy:privacy_check_packet` implementation, the result is stored in an accumulator so if a check has to be repeated it is just one map read.
 
-### Tracing hooks and handlers
+### Tracing
 
-`ejabberd_hooks:record` functions contain some (commented out for now) code which adds a record to the accumulator every time it is passed to a hook and every time it goes through a hook handler.
-This is very valuable for debugging.
+`origin_stanza` field is fully immutable for the lifespan of a single accumulator, so it's easier to correlate one of stanzas sent by a client with some "unexpected" stanza routed from completely different part of a server.
```

Comment:
`origin_stanza` field is fully immutable for the lifespan of a single accumulator, so it's easier to correlate one of the stanzas sent by a client with some "unexpected" stanza routed from a completely different part of a server.